libursa/src/signatures/secp256k1.rs

```diff
@@ -77,18 +79,19 @@ mod ecdsa_secp256k1sha256 {
             let pk = PublicKey(res.serialize().to_vec());
             Ok(pk)
         }
-        pub fn new() -> EcdsaSecp256k1Sha256Impl {
-            EcdsaSecp256k1Sha256Impl(libsecp256k1::Secp256k1::new())
+        pub fn new() -> Self {
+            Self(libsecp256k1::Secp256k1::new())
         }
-        pub fn keypair(&self, option: Option<KeyPairOption>) -> Result<(PublicKey, PrivateKey), CryptoError> {
+        pub fn keypair<D>(&self, option: Option<KeyPairOption>) -> Result<(PublicKey, PrivateKey), CryptoError> where D: Digest<OutputSize = U32> {
             let sk = match option {
                     Some(o) => {
                         match o {
                             KeyPairOption::UseSeed(seed) => {
                                 let mut s = [0u8; PRIVATE_KEY_SIZE];
                                 let mut rng = ChaChaRng::from_seed(*array_ref!(seed.as_slice(), 0, 32));
                                 rng.fill_bytes(&mut s);
```

Comment:
`s` will not be returned so the caller has no control to zero it? Should this code zero it or that is not a concern for now?